### PR TITLE
Implement pod label filter

### PR DESCRIPTION
**Pull request by michi-covalent**
2 commits:

- add pod_labels field.
- implement a filter that matches on pod_labels field.

### CODE REVIEW

File: cmd/protoc-gen-go-tetragon/eventchecker/field.go
Comment (sharlns) [inline cmd/protoc-gen-go-tetragon/eventchecker/field.go:213]: Is it enough the implement these checks later?

**Review by sharlns — APPROVED** [review]

File: cmd/protoc-gen-go-tetragon/eventchecker/field.go
Comment (willfindlay) [reply cmd/protoc-gen-go-tetragon/eventchecker/field.go:213]: @sharlns It's on my TODO list, but I don't think it has to block this PR. FWIW I think this is still WIP.

File: pkg/filters/labels.go
Comment (kkourt) [inline pkg/filters/labels.go:30]: I guess this means that if user specified 'key != label', which according to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors is legit, an event with no pod information will not match. I think that's (probaby) fine, but maybe we should document it?

(see other comment)

File: api/v1/tetragon/events.proto
Comment (kkourt) [inline api/v1/tetragon/events.proto:46]:
```
+ If  a labels is defined, events without Pod information will _always_ be filtered out.
```

(see other comment)

Comment (michi-covalent) [reply api/v1/tetragon/events.proto:46]: yeah let me add a separate documentation for the current filter syntax

File: pkg/filters/labels.go
Comment (michi-covalent) [reply pkg/filters/labels.go:30]: yeah that's an interesting point, let me think about this a bit and i'll document it 💭

File: api/v1/tetragon/events.proto
Comment (michi-covalent) [reply api/v1/tetragon/events.proto:46]: updated the comment. i'll follow up with a separate PR to document filter syntax once this PR gets merged.

**Review by kkourt — APPROVED** [review]

**Pull request merged**
